### PR TITLE
bazel: Add implementation of ABazelQuery

### PR DESCRIPTION
**Pull request by phlax**
Commit Message: bazel: Add implementation of ABazelQuery
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: tools/base/bazel_query.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+from functools import cached_property
+
+import abstracts
+
+from envoy.base.utils import ABazelQuery
+
+import envoy_repo
+
+
+@abstracts.implementer(ABazelQuery)
```
Comment (htuch) [inline tools/base/bazel_query.py:17]: This PR is clean, but it seems a bit magical. The reader doesn't know what ABazelQuery is, it's not in the repo etc. Also, where is the body of the implementation?

Not arguing at all with code structure, only with how the average reader is supposed to interpret and understand the code. What would a fresh reader do next after seeing this? How can we make the docs for ABazelQuery discoverable?

Comment (phlax) [reply tools/base/bazel_query.py:17]: yep, ill update with some info - ditto elsewhere

Comment (phlax) [reply tools/base/bazel_query.py:45]: updated above

File: tools/base/bazel_query.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+```console
+$ bazel run //tools/base:bazel_query "deps(source/...)" | jq "."
+```
+
+python usage:
+
+```python
+from tools.base.bazel_query import query
+
+result = query("source/...")
```
Comment (phlax) [inline tools/base/bazel_query.py:17]: `deps(source/...)`

**Review by htuch — APPROVED** [review]
LGTM, with the caveat again that hiding this `path` stuff from the wrapper is ideal.

**Pull request merged**